packages/plugin-backgroundcolor/src/BackgroundColor.ts

```diff
@@ -0,0 +1,164 @@
+import { Keymap } from '../../plugin-keymap/src/Keymap';
+import { Inline } from '../../plugin-inline/src/Inline';
+import { InlineNode } from '../../plugin-inline/src/InlineNode';
+import { getStyles, setStyles, setStyle, getStyle, removeStyle } from '../../utils/src/utils';
+import { VRange } from '../../core/src/VRange';
+import { Format } from '../../plugin-inline/src/Format';
+
+export interface ColorParams extends CommandParams {
+    color?: string; // css-valid color name
+}
+
+interface BackgroundColorConfig extends JWPluginConfig {
+    defaultBackgroundColor?: string;
+}
+
+const defaultConfiguration = {
+    defaultBackgroundColor: 'white',
+};
+
+export class BackgroundColor<T extends BackgroundColorConfig = BackgroundColorConfig> extends JWPlugin<T> {
+    static hasColor(node: VNode): boolean;
+    static hasColor(backgroundColor: string, node: VNode): boolean;
+    static hasColor(backgroundColor: string | VNode, node?: VNode): boolean {
+        if (backgroundColor instanceof VNode) {
+            node = backgroundColor;
+        }
+        const nodeBackgroundColor = getStyle(node, 'background-color');
+        if (!nodeBackgroundColor || backgroundColor instanceof VNode) {
+            return !!nodeBackgroundColor;
+        } else {
+            return nodeBackgroundColor === backgroundColor;
+        }
+    }
+    static dependencies = [Inline];
+    configuration = { ...defaultConfiguration, ...this.configuration };
+    commands = {
+        colorBackground: {
+            handler: this.colorBackground,
+        },
+        uncolorBackground: {
+            handler: this.uncolorBackground,
+        },
+    };
+    readonly loadables: Loadables<Keymap> = {
+        shortcuts: [
+            {
+                pattern: 'CTRL+H',
+                commandId: 'colorBackground',
+                // TODO: use dialog to get params
+                commandArgs: {
+                    color: 'yellow',
+                },
+            },
+            {
+                pattern: 'CTRL+SHIFT+H',
+                commandId: 'uncolorBackground',
+            },
+        ],
+    };
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Apply the given background color to the range.
+     *
+     * @param params
+     */
+    colorBackground(params: ColorParams): void {
+        const color = params.color;
+
+        if (params.context.range.isCollapsed()) {
+            // Set the style cache.
+            const inline = this.dependencies.get(Inline);
+            const currentCache = inline.styleCache || {};
+            inline.styleCache = { ...currentCache, 'background-color': color };
+        } else {
+            for (const node of this._highestSelectedNodes(params.context.range)) {
+                // Apply the style to the node or its first format.
```

Comment:
... so much for not having proper `modifiers` yet...

Reply:
Yup. Soon...